src/beamlime/handlers/data_reduction_handler.py

```diff
@@ -89,19 +110,26 @@ def make_handler(
         return PeriodicAccumulatingHandler(
             logger=self._logger,
             config=self._config,
-            preprocessor=ToNXevent_data(),
+            preprocessor=preprocessor,
             accumulators={f'reduced/{key.source_name}': accumulator},
         )
 
 
 class MultiplexingProxy(Accumulator[sc.DataArray, sc.DataGroup[sc.DataArray]]):
-    def __init__(self, stream_processors: list[StreamProcessor], key: Key) -> None:
+    def __init__(
+        self, stream_processors: list[StreamProcessor], key: Key, is_context: bool
+    ) -> None:
         self._stream_processors = stream_processors
         self._key = key
+        self._is_context = is_context
 
     def add(self, timestamp: int, data: sc.DataArray) -> None:
-        for stream_processor in self._stream_processors:
-            stream_processor.accumulate({self._key: data})
+        if self._is_context:
```

Comment:
Would there be more options in the future? (apart from `set_context` and `accumulate`?).
Is it worth making two different types of `MultiplexingProxy` that would do different things in their `add` method, instead of using the `is_context` boolean?

Reply:
I hope there will not be more. The proxy mechanism may change in the future. For example, I am currently working on supporting defining workflows and workflow params after service launch, which is currently not possible (but required, unless we want users to restart the service just to change reduction parameters). So this code is being touched/changed anyway.